### PR TITLE
DR-3022: Page layout component and metadata

### PR DESCRIPTION
**Pull request by 7emansell**
## Ticket:

Resolves JIRA ticket [DR-3022](https://newyorkpubliclibrary.atlassian.net/browse/DR-3022).

## This PR does the following:
- Creates custom (client side) page layout component with `activePage`, `breadcrumbs` and `DSProvider`, `FeedbackBox`, etc.
   - Moves Adobe Analytics into this as well
- Moves home and about page contents into client components
- Adds `PageLayout` to all pages' corresponding client component
- Adds Next's `Metadata` to all pages (bc now they're server components!)
   - Dynamically generating metadata title tag from url on `/collections/lane/[slug]` and `/divisions/[slug]`– also using this title for the breadcrumbs. Once we are actually fetching page data, this won't be how we generate the breadcrumbs. Or maybe it'll be some hybrid of these, whatever performs better
   - Statically generating title tag on `/collections` and `/divisions`
- Adds router mock to home and about pages' tests

## How has this been tested?

Locally (necessary to see breadcrumbs)

### Checklist:

- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.
- [x] I have updated the CHANGELOG.md.

### CODE REVIEW

File: app/about/page.tsx [hunk 1]
```diff
@@ -1,35 +1,10 @@
-"use client";
-import { Box } from "@nypl/design-system-react-components";
-import aboutPageElements from "../data/aboutPageElements";
-import React from "react";
+import AboutPage from "app/components/aboutPage/aboutPage";
+import { Metadata } from "next";
```
Comment (avertrees) [inline app/about/page.tsx:2]: wait - I thought Metadata only worked on server components. Does this work with client components too or does this mean we're using server components now? (eyes emoji)

Comment (7emansell) [reply app/about/page.tsx:2]: read PR description <3

Comment (avertrees) [reply app/about/page.tsx:2]: I did and that's why I left a comment but I'm still confused! More surprised, actually. This feels like a big deal?

Comment (avertrees) [reply app/about/page.tsx:2]: I thought we couldn't use the DS with server components

Comment (avertrees) [reply app/about/page.tsx:2]: nvm, jumped the gun a bit. I see how this works now

**Review by avertrees — APPROVED** [review]

File: __tests__/pages/aboutpage.test.tsx [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+  useRouter: jest.fn(),
+  usePathname: jest.fn(),
+}));
+
 describe("About page accessibility", () => {
   it("passes axe accessibility test", async () => {
     const { container } = render(<About />);
     expect(await axe(container)).toHaveNoViolations();
-  }, 15000);
+  }, 60000);
```
Comment (EdwinGuzman) [inline __tests__/pages/aboutpage.test.tsx:15]: Oiy, that's a big timeout increase.

**Review by EdwinGuzman — COMMENTED** [review]
How do you feel about making a directory under `components` named `pages` for the about and home page files? Later on it can include the other page files. I can see it becoming a big folder with subfolders for each page. It might or might not be an actual problem but if it does, you can always make the change later.

File: app/collections/lane/[slug]/page.tsx [hunk 3]
```diff
@@ -1,10 +1,36 @@
...
+import PageLayout from "app/components/pageLayout/pageLayout";
 
-export default function Lane() {
+type LaneProps = {
+  params: { slug: string };
+};
+
+export async function generateMetadata(
+  { params }: LaneProps,
+  parent: ResolvingMetadata
```
Comment (EdwinGuzman) [inline app/collections/lane/[slug]/page.tsx:11]: Do you need the parent arg in the `generateMetadata` functions?

File: app/components/pageLayout/pageLayout.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+}: PropsWithChildren<PageLayoutProps>) => {
+  const pathname = usePathname();
+  // Track page view events to Adobe Analytics
+  useEffect(() => {
+    /*
+    "asPath has been removed because the concept of as has been removed from the new router."
+    https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration
+    */
+    console.log("pathname", pathname);
+    const route = pathname ? pathname : "";
```
Comment (EdwinGuzman) [inline app/components/pageLayout/pageLayout.tsx:36]: You should also be able to do `pathname || ""`

Comment (EdwinGuzman) [reply app/components/pageLayout/pageLayout.tsx:36]: Or perhaps `??` instead.

File: app/components/pageLayout/pageLayout.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  breadcrumbs,
+}: PropsWithChildren<PageLayoutProps>) => {
+  const pathname = usePathname();
+  // Track page view events to Adobe Analytics
+  useEffect(() => {
+    /*
+    "asPath has been removed because the concept of as has been removed from the new router."
+    https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration
+    */
+    console.log("pathname", pathname);
```
Comment (EdwinGuzman) [inline app/components/pageLayout/pageLayout.tsx:35]: Remove logs

File: app/components/pageLayout/pageLayout.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    const route = pathname ? pathname : "";
+    trackVirtualPageView(route);
+  });
+  const [view, setView] = React.useState("form");
+  const { onOpen, isOpen, onClose, FeedbackBox } = useFeedbackBox();
```
Comment (EdwinGuzman) [inline app/components/pageLayout/pageLayout.tsx:40]: Not part of this PR but `onOpen` is not being used.

**Review by EdwinGuzman — APPROVED** [review]
Great to see app router features!

As an aside: the DS needs to make an update to the `Breadcrumbs` component to allow the Next.js `Link` component instead of a regular link. It's currently making full page refreshes which is not a problem but it'd be nice to use the in-app router.

Aside 2: I see this warning in the console
`VM26917:106 NYPL Reservoir FeaturedContent: A value for 'imageProps.alt' is required. `  but it's also in prod so don't fix this now. Create a ticket to clean up the warnings that get logged.

**Comment by EdwinGuzman** [comment]
Ticket for breadcrumbs routing https://newyorkpubliclibrary.atlassian.net/browse/DSD-1797

File: __tests__/pages/aboutpage.test.tsx (hunk 2 above)
Comment (7emansell) [reply __tests__/pages/aboutpage.test.tsx:15]: They don't take this long locally but I just want the test hook to not fail bc of timeout
